server/src/graphql/mutations/new_transaction.rs

```diff
@@ -0,0 +1,170 @@
+use super::super::nodes::{Squad, Transaction};
+use crate::db::{
+    models,
+    schema::{balance, node, squad, txn, txn_part},
+    Pool,
+};
+use anyhow::Result;
+use async_graphql::{
+    validators::{InputValueValidator, IntNonZero, ListMinLength},
+    FieldError, FieldResult, Value, ID,
+};
+use diesel::prelude::*;
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use tokio_diesel::*;
+use uuid::Uuid;
+
+struct ChangesSumToZero {}
+
+impl InputValueValidator for ChangesSumToZero {
+    fn is_valid(&self, value: &Value) -> Option<String> {
```

Comment:
Wait why do we use the rust Option type for this instead of a boolean lol

Reply:
They also seem backwards... `None` seems like it would be `false`, no?

Reply:
I guess we don't have a choice with the diesel attributes...

Reply:
Yeah I don't like this design either but it's mandated by the async_graphql API. The good news is that in the newer version of the API (with the other PR adopts) it's changed to `Result<(), String>`, so you're either returning `Ok(())` or `Err(error_message_string)`, which is a lot better

Reply:
that's good lol, that's the way every other language does it xD